src/scalar.rs

```diff
@@ -478,62 +477,31 @@ impl PrimeField for Scalar {
     }
 }
 
-#[cfg(not(target_pointer_width = "64"))]
-type ReprBits = [u32; 8];
-
-#[cfg(target_pointer_width = "64")]
-type ReprBits = [u64; 4];
-
 impl PrimeFieldBits for Scalar {
     // Representation in non-Montgomery form.
-    type ReprBits = ReprBits;
+    type ReprBits = [u8; 32];
```

Comment:
this change doesn't quite work for us, as I am trying to mirror this: https://github.com/zkcrypto/bls12_381/blob/main/src/scalar.rs#L729